File: algo-armstrong-numbers/js/armstrongNumbers.js
// How can you make this more scalable and reusable later?
const isArmstrongNumber = function (number) {
  if (number >= 0 && number < 10) {
    return true;
  }
  const digits = number.toString().split("").map(Number);
  let power = digits.length;
  let total = 0;
  for (let digit of digits) {
    total += digit ** power;
  }
  return total === number;
};

exports.findArmstrongNumbers = function (numbers) {
  const armstrongNumbers = [];
  for (let num of numbers) {
    if (isArmstrongNumber(num)) {
      armstrongNumbers.push(num);
    }
  }
  return armstrongNumbers;
};

File: algo-armstrong-numbers/python/armstrong_numbers.py
# How can you make this more scalable and reusable later?
def is_armstrong_number(num):
    if num < 10:
        return True
    digits = [int(n) for n in str(num)]
    power = len(digits)
    total = 0
    for d in digits:
        total += d**power
    return total == num


def find_armstrong_numbers(numbers):
    armstrong_numbers = []
    for num in numbers:
        if is_armstrong_number(num):
            armstrong_numbers.append(num)
    return armstrong_numbers
